### PR TITLE
Renaming opendistro security plugin name and Jar

### PR DESCRIPTION
**Pull request by cliu123**
##  opendistro-for-elasticsearch/security pull request intake form
_Please provide as much details as possible to get feedback/acceptance on your PR quickly_

 
 1. __Category:__ (_Enhancement, New feature, Bug fix, Test fix, Refactoring, Maintenance, Documentation_)
 Maintenance
 
 
 2. __Github Issue # or road-map entry, if available:__

 3. __Description of changes:__
- Rename the following:
    * name of the plugin in the plugin descriptor
    * folder name of the security plugin after installation
    * plugin jar name
    * maven artifacts name
    * output of ES _cat/plugins and similar API that includes plugin name

- The following will stay unchanged:
    * security plugin system index
    * security plugin REST API

 4. __Why these changes are required?__ 
Keep name of the plugin consitent with other ODFE plugins.

 5. __What is the old behavior before changes and new behavior after changes?__ (_Please add any example/logs/screen-shot if available_)
No changes in behavior.

 6. __Testing done:__ (_Please provide details of testing done: Unit testing, integration testing and manual testing_)
 UT and MT
Tested following cases:
- folder name of the security plugin after installation
- plugin jar name
- maven artifacts name
- output of ES _cat/plugins
- security configuartions are loaded into plugins/opendistro-security/securityconfig and elasticsearch.yml
 
 7. __TO-DOs, if any:__ (_Please describe pending items and provide Github issues# for each of them_)

 8. __Is it backport from main branch?__ (_If yes, please add backport PR # and commits #_)

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

### CODE REVIEW

**Review by debjanibnrj — APPROVED** [review]

File: tools/install_demo_configuration.sh [hunk 1]
```diff
@@ -186,7 +186,7 @@ echo "Elasticsearch lib dir: $ES_LIB_PATH"
 echo "Detected Elasticsearch Version: $ES_VERSION"
 echo "Detected Open Distro Security Version: $SECURITY_VERSION"
 
-if $SUDO_CMD grep --quiet -i opendistro_security "$ES_CONF_FILE"; then
+if $SUDO_CMD grep --quiet -i opendistro-security "$ES_CONF_FILE"; then
```
Comment (vrozov) [inline tools/install_demo_configuration.sh:189]: opendistro security settings are not renamed

Comment (cliu123) [reply tools/install_demo_configuration.sh]: Yes, we're not renaming the configurations. Chaned it back.

**Review by vrozov — CHANGES_REQUESTED** [review]
Instead of hardcoding plugin name into strings, please introduce a constant (can be separate for java and scripts) that defines plugin name.

File: src/main/java/com/amazon/opendistroforelasticsearch/security/configuration/ConfigurationRepository.java [hunk 2]
```diff
@@ -123,8 +123,9 @@ public void run() {
                     if(installDefaultConfig.get()) {
 
                         try {
+                            final String pluginName = "opendistro-security";
```
Comment (vrozov) [inline src/main/java/com/amazon/opendistroforelasticsearch/security/configuration/ConfigurationRepository.java:126]: Must be a constant on plugin name.

File: build.gradle [hunk 3]
```diff
@@ -31,7 +31,7 @@ ospackage {
 
     into '/usr/share/elasticsearch/plugins'
     from(zipTree(project.file(archivePath).absolutePath)) {
-        into "opendistro_security"
+        into "opendistro-security"
```
Comment (vrozov) [inline build.gradle:34]: Is it necessary to make changes on line 56 as well?

File: tools/install_demo_configuration.sh [hunk 4]
```diff
@@ -165,16 +165,16 @@ fi
 ES_CONF_DIR=$(dirname "${ES_CONF_FILE}")
 ES_CONF_DIR=`cd "$ES_CONF_DIR" ; pwd`
 
-if [ ! -d "$ES_PLUGINS_DIR/opendistro_security" ]; then
+if [ ! -d "$ES_PLUGINS_DIR/opendistro-security" ]; then
```
Comment (vrozov) [inline tools/install_demo_configuration.sh:168]: Needs to be replaced with a constant

**Review by vrozov — CHANGES_REQUESTED** [review]

File: tools/install_demo_configuration.sh (hunk 4 above)
Comment (cliu123) [reply tools/install_demo_configuration.sh]: Done.

File: src/main/java/com/amazon/opendistroforelasticsearch/security/OpenDistroSecurityPlugin.java [hunk 5]
```diff
@@ -200,6 +200,7 @@
     private volatile NamedXContentRegistry namedXContentRegistry = null;
     private volatile DlsFlsRequestValve dlsFlsValve = null;
     private volatile Salt salt;
+    public static final String pluginName = "opendistro-security";
```
Comment (vrozov) [inline src/main/java/com/amazon/opendistroforelasticsearch/security/OpenDistroSecurityPlugin.java:203]: nit: constant use all capital: `PLUGIN_NAME`. Move it to line 182 where other `static` are defined.

Comment (cliu123) [reply src/main/java/com/amazon/opendistroforelasticsearch/security/OpenDistroSecurityPlugin.java]: Updated

File: src/main/java/com/amazon/opendistroforelasticsearch/security/configuration/ConfigurationRepository.java (hunk 2 above)
Comment (cliu123) [reply src/main/java/com/amazon/opendistroforelasticsearch/security/configuration/ConfigurationRepository.java]: Done

File: build.gradle (hunk 3 above)
Comment (cliu123) [reply build.gradle]: Renamed on line 56. And resued the `packageName` defined in the script.

File: src/main/java/com/amazon/opendistroforelasticsearch/security/configuration/ConfigurationRepository.java [hunk 6]
```diff
@@ -124,7 +125,7 @@ public void run() {
 
                         try {
                             String lookupDir = System.getProperty("security.default_init.dir");
-                            final String cd = lookupDir != null? (lookupDir+"/") : new Environment(settings, configPath).pluginsFile().toAbsolutePath().toString()+"/opendistro_security/securityconfig/";
+                            final String cd = lookupDir != null? (lookupDir+"/") : new Environment(settings, configPath).pluginsFile().toAbsolutePath().toString()+"/" + OpenDistroSecurityPlugin.PLUGIN_NAME + "/" + "securityconfig/";
```
Comment (palashhedau) [inline src/main/java/com/amazon/opendistroforelasticsearch/security/configuration/ConfigurationRepository.java:128]: Did you test this ? Is this working as expected?

File: tools/install_demo_configuration.sh [hunk 7]
```diff
@@ -165,16 +166,16 @@ fi
...
   exit -1
 fi
 
 ES_VERSION=("$ES_LIB_PATH/elasticsearch-*.jar")
 ES_VERSION=$(echo $ES_VERSION | sed 's/.*elasticsearch-\(.*\)\.jar/\1/')
 
-SECURITY_VERSION=("$ES_PLUGINS_DIR/opendistro_security/opendistro_security-*.jar")
-SECURITY_VERSION=$(echo $SECURITY_VERSION | sed 's/.*opendistro_security-\(.*\)\.jar/\1/')
+SECURITY_VERSION=("$ES_PLUGINS_DIR/$PLUGIN_NAME/opendistro-security-*.jar")
+SECURITY_VERSION=$(echo $SECURITY_VERSION | sed 's/.*opendistro-security-\(.*\)\.jar/\1/')
```
Comment (peterzhuamazon) [inline tools/install_demo_configuration.sh:178]: Why leave the jar file name hardcoded instead of using the `$PLUGIN_NAME` variable you created earlier?

**Review by palashhedau — DISMISSED** [review]

File: src/main/java/com/amazon/opendistroforelasticsearch/security/configuration/ConfigurationRepository.java (hunk 6 above)
Comment (cliu123) [reply src/main/java/com/amazon/opendistroforelasticsearch/security/configuration/ConfigurationRepository.java:128]: Tested this and confirmed with Palash. The security plugin configurations have been successfully loaded.

File: tools/install_demo_configuration.sh (hunk 7 above)
Comment (cliu123) [reply tools/install_demo_configuration.sh:178]: This is actually not plugin name. This is the artifact name.

File: src/main/java/com/amazon/opendistroforelasticsearch/security/tools/AuditConfigMigrater.java [hunk 8]
```diff
@@ -48,7 +48,7 @@
 
     public static void main(String[] args) {
         options.addOption(Option.builder("s").argName("source").hasArg().desc("Path to elasticsearch.yml file to migrate. If not specified, will try to lookup env " + ES_PATH_CONF_ENV + " followed by lookup in current directory.").build());
-        options.addOption(Option.builder("oad").argName("output-audit-dir").hasArg().desc("Output directory to store the generated " + AUDIT_YML + " file. To be uploaded in the index, the file must be present in plugins/opendistro_security/securityconfig/ or use securityadmin tool.").build());
+        options.addOption(Option.builder("oad").argName("output-audit-dir").hasArg().desc("Output directory to store the generated " + AUDIT_YML + " file. To be uploaded in the index, the file must be present in plugins/opendistro-security/securityconfig/ or use securityadmin tool.").build());
```
Comment (hsiang9431-amzn) [inline src/main/java/com/amazon/opendistroforelasticsearch/security/tools/AuditConfigMigrater.java:51]: Can you also use the public constant `PLUGIN_NAME` defined in `src/main/java/com/amazon/opendistroforelasticsearch/security/OpenDistroSecurityPlugin.java`

**Review by hsiang9431-amzn — COMMENTED** [review]
Looks cool to me. Just left a comment for slight improvement

**Comment by vrozov** [comment]
Please revert changes to `maven-assembly-plugin` configuration done as part of previous commit. Review all previous commits related to the rename and see if they are still necessary.

**Review by hsiang9431-amzn — APPROVED** [review]

**Comment by cliu123** [comment]
The PR is no longer needed.

**Pull request closed without merging**